### ISSUE TITLE
 invalid value encountered in less

### ISSUE THREAD

**Issue by DMTSource** [awjuliani/DeepRL-Agents#27]
Sometimes when working with new levels or environments I get the error:

RuntimeWarning: invalid value encountered in less
a = np.random.choice(a_dist[0],p=a_dist[0])

I'm guessing there is a nan in the nerwork? This usually occurs after many steps. I have tried strong and weak gradient clipping methods and my rewards are always within a small range such as ~(0-1). My gradient norms tends to grow with time and soar past the clipping value.

Any suggestions for how to prevent this issue with the network?

**Comment by DMTSource** [comment]
I have tried ensuring that any log math has a tiny constant added to it to ensure no bad math is occurring in the loss & entropy calculations, but the problem still persists. I think the issue might be more related to initialization. If I switch to RMSProp optimzation then i can only get a few steps into the training before nans appear; even with a tiny learning rate and very strong gradient clipping.

**Comment by bluemoon93** [comment]
It is simple to solve this problem. Just use the line below instead of `tf.log(self.policy)` when calculating `self.entropy`.
`self.log_policy = tf.log(tf.clip_by_value(self.policy, 1e-20, 1.0))         # avoid NaN with clipping when value in policy becomes zero`
